packages/commerce-sdk-react/src/hooks/ShopperCustomers/mutation.test.tsx

```diff
@@ -0,0 +1,233 @@
+                            name: /update customer/i
+                        })
+                    )
+
+                    // CACHE Pre-populate cache with the query keys we invalidate on success
+                    queryClient.setQueryData(
+                        ['/customers', CUSTOMER_ID, '/payment-instruments'],
+                        {}
+                    )
+                    queryClient.setQueryData(['/customers', CUSTOMER_ID, '/addresses'], {})
+                    queryClient.setQueryData(['/customers', '/external-profile'], {})
+
+                    expect(
+                        queryClient.getQueryState([
+                            '/customers',
+                            CUSTOMER_ID,
+                            '/payment-instruments'
+                        ])?.isInvalidated
+                    ).toBeFalsy()
+                    expect(
+                        queryClient.getQueryState(['/customers', CUSTOMER_ID, '/addresses'])
+                            ?.isInvalidated
+                    ).toBeFalsy()
+                    expect(
+                        queryClient.getQueryState(['/customers', '/external-profile'])
+                            ?.isInvalidated
+                    ).toBeFalsy()
+
+                    // Mocking the server request
+                    nock('http://localhost:3000')
+                        .patch((uri) => {
+                            return uri.includes('/customer/shopper-customers/')
+                        })
+                        .reply(200, {
+                            customerId: CUSTOMER_ID,
+                            email: CUSTOMER_EMAIL,
+                            login: CUSTOMER_EMAIL
+                        })
+
+                    const button = screen.getByRole('button', {
+                        name: /update customer/i
+                    })
+
+                    fireEvent.click(button)
+                    await waitFor(() => screen.getByText(/email/i))
+                    expect(screen.getByText(/email/i)).toBeInTheDocument()
+
+                    // TODO: Remove DEBUG console.log printing queries in cache
+                    // console.log('queryClient.getQueryCache():',queryClient.getQueryCache())
+
+                    // CACHE assert updated query keys
+                    expect(
+                        queryClient.getQueryState([
+                            '/customers',
+                            CUSTOMER_ID,
+                            {customerId: CUSTOMER_ID}
+                        ])?.isInvalidated
+                    ).toBeFalsy()
+
+                    // CACHE assert invalidated query keys
+                    expect(
+                        queryClient.getQueryState([
+                            '/customers',
+                            CUSTOMER_ID,
+                            '/payment-instruments'
+                        ])?.isInvalidated
+                    ).toBeTruthy()
```

Comment:
Ok, so we're testing the before and after the Update Customer button is clicked 👍 
- before: all queries should not be invalidated
- after: all but one query are invalidated